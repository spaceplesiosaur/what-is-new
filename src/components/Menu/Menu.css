.menuContainer {
  width: 25em;
  display: flex;
  flex-direction: column;
  text-align: center;
  background-color: lavender;
}

.menuHeader {
  color: black;
  font-size: 2em;
  font-weight: bold;
  margin-top: .5em;
}

.menuCategory {
  display: flex;
  flex-direction: row;
  margin: 1.5em;
}

.menuCategory:hover {
  background-color: yellow;
}

.menuItemName {
  color: black;
}

img {
  height: 20px;
  width: 20px;
  margin-right: .3em;
}
