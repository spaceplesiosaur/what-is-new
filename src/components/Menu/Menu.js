import React from 'react';
import './Menu.css'
import { FaBeer } from 'react-icons/fa';

const Menu = (props) => {
  const menuOptions = props.menuItems
  const listItems = menuOptions.map((item, i) => {
    const topicPicker = () => props.setTopic(item.id)
    return (
        <li className="menuCategory" key={i} id={item.id} onClick={topicPicker}>
          <img className="menuItemImage" alt="menu icon" src={item.image}/>
          <p className="menuItemName">{item.name}</p>
        </li>
      )
  });
  return (
      <section className="menuContainer">
        <h1 className="menuHeader">What's New?</h1>
        <ul>
          {listItems}
        </ul>
      </section>
    )
}


export default Menu;
