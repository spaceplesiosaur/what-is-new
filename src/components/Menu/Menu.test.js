import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should match the snapshot with correct data', () => {
    const wrapper = shallow(<Menu
    menuItems={["test"]}
    setTopic={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
