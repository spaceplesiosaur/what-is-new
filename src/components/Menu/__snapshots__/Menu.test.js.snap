// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with correct data 1`] = `
<section
  className="menuContainer"
>
  <h1
    className="menuHeader"
  >
    What's New?
  </h1>
  <ul>
    <li
      className="menuCategory"
      key="0"
      onClick={[Function]}
    >
      <img
        alt="menu icon"
        className="menuItemImage"
      />
      <p
        className="menuItemName"
      />
    </li>
  </ul>
</section>
`;
